src/abc_atlas_access/abc_atlas_cache/manifest.py

```diff
@@ -0,0 +1,284 @@
+                    "metadata sub-directory found."
+                )
+            output_data_list.extend(list(sub_directories['metadata'].keys()))
+        else:
+            for sub_dir in sub_directories.keys():
+                if sub_dir == 'metadata':
+                    continue
+                for sub_dir_key in sub_directories[sub_dir].keys():
+                    # Check for files with multiple kinds (e.g. raw and log2)
+                    if "files" in sub_directories[sub_dir][sub_dir_key].keys():
+                        output_data_list.append(sub_dir_key)
+                    else:
+                        output_data_list.extend(
+                            ["%s/%s" % (sub_dir_key, key)
+                             for key in sub_directories[sub_dir][
+                                 sub_dir_key].keys()]
+                        )
+        output_data_list.sort()
+        if len(output_data_list) == 0 and is_metadata:
+            raise KeyError(
+                f"No metadata files found in directory {directory}. Metadata "
+                "sub-directory is empty."
+            )
+        elif len(output_data_list) == 0 and not is_metadata:
+            raise KeyError(
+                f"No data files found in directory {directory}."
+            )
+        return output_data_list
+
+    @property
+    def data(self):
+        """
+        The raw dictionary data of the manifest file.
+        """
+        return self._data
+
+    @property
+    def version(self):
+        """
+        The version of the dataset currently loaded
+        """
+        return self._version
+
+    @property
+    def resource_uri(self):
+        """
+        The URI of the dataset currently loaded
+        """
+        return self._resource_uri
+
+    @property
+    def list_directories(self):
+        """
+        List of all directories that are part of the dataset.
+        """
+        return self._directory_list
+
+    def list_metadata_files(self, directory: str) -> List[str]:
+        """
+        List all metadata files in the specified directory.
+
+        Parameters
+        ----------
+        directory: str
+            The directory to list metadata files in.
+
+        Returns
+        -------
+        list of str
+            List of all metadata files in the specified directory.
+        """
+        return self._list_data_in_directory(directory, is_metadata=True)
+
+    def list_data_files(self, directory: str) -> List[str]:
+        """
+        List all data files that are not metadata in the specified directory.
+
+        Parameters
+        ----------
+        directory: str
+            The directory to list data files in.
+
+        Returns
+        -------
+        list of str
+            List of all data files that are not metadata in the specified
+            directory.
+        """
+        return self._list_data_in_directory(directory, is_metadata=False)
+
+    def get_file_attributes(
```

Comment:
How will users interact with this method? Will it be something like
```
file_attr = manifest.get_file_attributes(...)

cache_obj.download(file_attr)
```
if that is the case, I wonder if this method should have a more explanatory name like `get_download_specification`. "File attributes" sounds too generic (I acknowledge you probably inherited that name from an earlier version of me...).

Reply:
Users won't interact with this method. They will interact with the cache directly via a method that will look like this. The method will then return a pandas df (for metadata) or a path to the h5ad/other file and let them open it themselves.

Reply:
That's fine. No changes, then.